### PR TITLE
Fix for physical propagation

### PR DESCRIPTION
**Pull request by ojdf**
At the moment downward physical propagation is broken with more than one phase screen. This has fixed it for me, and also (maybe) not broken upward propagation

### CODE REVIEW

File: soapy/lineofsight.py [hunk 1]
```diff
@@ -400,8 +400,8 @@ def physical_atmosphere_propagation(
     else:
         ht = layer_altitudes[scrnNo-1]
         ht_final = 0
-        scrnRange = scrnRange[::-1]
         scrnAlts = layer_altitudes[::-1]
+        phase_screens = phase_screens[::-1]
```
Comment (matthewtownson) [inline soapy/lineofsight.py]: This will cause the order of phase screens to swap every iteration, keep the inverted height screens as a separate variable

**Comment by matthewtownson** [comment]
Great spot, but the current fix won't work in a full AO simulation, if you could fix the phase_screens variable I think it should be OK to merge.

**Comment by andrewpaulreeves** [comment]
Hi chaps, thanks for the good work @ofdf, your definitely onto something with it, though I think @matthewtownson is right that it still isn't quite right.
Apologies for my late reply, but I'd quick like to do a quick test before we merge and make sure we get it right! I'll try to check it out tonight or tomorrow. 
Andrew

**Comment by ojdf** [comment]
ok the propagation is now done on a copy of the phase screens that are passed to the function.

I've also cleaned up the commits...

**Comment by andrewpaulreeves** [comment]
OK - I've had a test and I think this is all good to go....except actually @matthewtownson suggestion of the array copy probably isn't needed (soz @matthewtownson ). We do get a "view" of the phase screens back, but they never actually get changed and the "phase_screens" variable is only scoped in the propagation function so won't affect the los.phase_screens attribute at all (I think this is all the case...).

Not a major issue as the ASP algorithm will dominate performance wise anyway, but less unneeded lines of code is always better.

**Comment by ojdf** [comment]
ok, I've reverted that commit

**Pull request merged**